src/plugins/controls/public/range_slider/components/range_slider_popover.tsx

```diff
@@ -26,9 +27,11 @@ import { rangeSliderReducers } from '../range_slider_reducers';
 import { RangeSliderReduxState } from '../types';
 import { RangeSliderStrings } from './range_slider_strings';
 
-export const RangeSliderPopover: FC = () => {
+// Unfortunately, wrapping EuiDualRange in `withEuiTheme` has created this annoying/verbose typing
```

Comment:
Is there anything we can do on our side to get rid of these complex types? Maybe a forward ref? Or some other way of getting a handle on the class?

Reply:
Honestly? Without knowing exactly why you're calling the internal `EuiDualRange`'s internal `onResize` class method... my recommendation is to **not** do that 😅 (which removes the need for the `ref` entirely, and thus the ref 
typing shenanigans).

If we ever convert `EuiRange`/`EuiDualRange` to functional non-class components as a tech debt item, you'd lose access to this API entirely (it's not something we're deliberately exposing or documenting as a consumer API). I did [create a regression test](https://github.com/elastic/eui/pull/6467) to ensure we continued exposing `onResize` via ref, just for this specific Kibana usage of it, but honestly, we'd prefer y'all not reach for it at all, or figure out what problem it's solving and have us try to solve it in a less technically hacky way.

I don't have the context as to why that original call was added or what it's doing - but it's definitely a code smell that makes me think either this usage of it is trying to do things EuiDualRange is not meant to do, or is super custom and undoing things EuiDualRange does and then re-doing them later (also not great).

Reply:
If I remember correctly, this usage is due to us not being able to use `EuiInputPopover`, but still wanting to have the popover to be tied to the width of the input. I can't exactly remember why we couldn't use `EuiInputPopover` though.

I wonder if we would run into any regressions by removing that call entirely? @cqliu1, do you remember why we had to use `EuiPopover`?